### PR TITLE
chore(workflow): add typecheck to pr validation

### PR DESCRIPTION
**Pull request by JackHamer09**
# What :computer: 
* Added typecheck to PR validation workflow

# Why :hand:
* To make sure there are no typescript errors in the code

# Evidence :camera:
image

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -10,6 +10,7 @@
     "build": "tsc -p .",
     "replaceVars": "ts-node --transpile-only src/replace.ts bin/analytics.js",
     "test": "echo \"Error: no test specified\" && exit 1",
+    "typecheck": "tsc -p . --noEmit",
```
Comment (MexicanAce) [inline package.json:13]: Should this be included in the `lint` script below? I figured type-checking is a part of linting.

**Review by dutterbutter — DISMISSED** [review]

File: package.json (hunk 1 above)
Comment (JackHamer09) [reply package.json:13]: Makes sense

Comment (JackHamer09) [reply package.json:13]: I added it separately because to properly use eslint and typecheck for `lint-staged` plugin it needed to be separate

**Review by MexicanAce — APPROVED** [review]
LGTM

**Pull request merged**